modules/emulator/R/minimize.GP.R

```diff
@@ -211,33 +212,45 @@ mcmc.GP <- function(gp, x0, nmcmc, rng, format = "lin", mix = "joint", splinefcn
   for (g in start:nmcmc) {
     
     if (mix == "joint") {
-      ## propose new
-      xnew <- xcurr
+
+      # adapt
       if ((g > 2) && ((g - 1) %% settings$assim.batch$jump$adapt == 0)) {
         params.recent <- samp[(g - settings$assim.batch$jump$adapt):(g - 1), ]
         colnames(params.recent) <- names(x0)
         # accept.count <- round(jmp@arate[(g-1)/settings$assim.batch$jump$adapt]*100)
         jcov <- pda.adjust.jumps.bs(settings, jcov, accept.count, params.recent)
         accept.count <- 0  # Reset counter
       }
+      
+      ## propose new
       repeat {
         xnew <- mvrnorm(1, unlist(xcurr), jcov)
         if (bounded(xnew, rng)) {
           break
         }
       }
       # if(bounded(xnew,rng)){
-      currY <- get.y(gp, xcurr, n.of.obs, llik.fn, priors, settings)
+      
+      # update model parameters conditioned on likelihood parameters
+      # when llik.par (= currll) is passed get.y returns posterior value
+      currY <- get.y(gp, xcurr, n.of.obs, llik.fn, priors, settings, currll)
       ycurr <- currY$posterior.prob
-      pcurr <- currY$par
-      newY  <- get.y(gp, xnew, n.of.obs, llik.fn, priors, settings, currY$llik.par)
+
+      newY  <- get.y(gp, xnew, n.of.obs, llik.fn, priors, settings, currll)
       ynew  <- newY$posterior.prob
+
       if (is.accepted(ycurr, ynew)) {
         xcurr <- xnew
-        pcurr <- newY$par
-        
+        #pcurr <- newY$par
+        #curll <- newY$llik.par
         accept.count <- accept.count + 1
       }
+      
+      # update likelihood parameters conditioned on model parameters
+      # when llik.par (= currll) is not passed, get.y updates ll-params, given model parameters
+      # and returns llik.par
+      currll <- get.y(gp, xcurr, n.of.obs, llik.fn, priors, settings)
```

Comment:
This is what I was worried you were doing when you made get.y polymorphic above. In addition to being confusing, this design ends up making a call to predict.gp and a random draw of SS that's different from the one used in the M-H algorithm. That is both incorrect and an unnecessary computation. What would be much simpler would be to drop the polymorphism in get,y (the return if is.null(llik.par), which should never happen), have get.y return SS as an additional arg, and then update SS in the M-H accept step as noted earlier.  By contrast, here I think all you need is
```
## update statistical parameters by Gibbs sampling
currll <- pda.calc.llik.par(settings, n.of.obs, SScurr)
```